File: analyze_results.js
const fs = require('fs');

const data = JSON.parse(fs.readFileSync('g:/IBM Product/job-d6rg8aropkic73fj56c0-result.json', 'utf8'));
const samples = data.results[0].data.meas.samples;

const counts = {};
samples.forEach(s => {
    counts[s] = (counts[s] || 0) + 1;
});

const total = samples.length;
const sorted = Object.entries(counts).sort((a, b) => b[1] - a[1]);

const summary = sorted.map(([hex, count]) => {
    const val = parseInt(hex, 16);
    const bitstring = val.toString(2).padStart(4, '0');
    return {
        hex,
        bitstring,
        count,
        probability: ((count / total) * 100).toFixed(2) + '%'
    };
});

console.log(JSON.stringify(summary, null, 2));

File: analyze_results.py
import json
from collections import Counter

with open('g:/IBM Product/job-d6rg8aropkic73fj56c0-result.json', 'r') as f:
    data = json.load(f)

samples = data['results'][0]['data']['meas']['samples']
counts = Counter(samples)
total = len(samples)

results = []
for hex_val, count in counts.most_common():
    val = int(hex_val, 16)
    bitstring = bin(val)[2:].zfill(4)
    prob = count / total
    results.append({
        "hex": hex_val,
        "bitstring": bitstring,
        "count": count,
        "probability": f"{prob:.2%}"
    })

print(json.dumps(results, indent=2))
